### PR TITLE
Cache StarredFiles

### PR DESCRIPTION
**Pull request by forouher**
Probably not a very important feature inside the Seafile app.
However the SeafileProvider (next pull request) relies on it.

### CODE REVIEW

**Comment by lins05** [comment]
To make the cache truly useful, we should use the cached starred files list, and only send a `LoadStarredFilesTask` after a cache timeout, like 5 or 10 minutes.

**Comment by forouher** [comment]
@lins05 yes, I didn't think that through. Should be easy to do, give me a day or two.

**Comment by lins05** [comment]
LGTM. @Logan676 Please test this.

File: src/com/seafile/seadroid2/ui/fragment/StarredFragment.java [hunk 1]
```diff
@@ -133,7 +133,13 @@ public void refreshView() {
             mPullRefreshListView.onRefreshComplete();
             Toast.makeText(mActivity, getString(R.string.network_down), Toast.LENGTH_SHORT).show();
         }
-        ConcurrentAsyncTask.execute(new LoadStarredFilesTask(getDataManager()));
+        List<SeafStarredFile> starredFiles = getDataManager().getCachedStarredFiles();
+        boolean refreshTimeout = getDataManager().isStarredFilesRefreshTimeout();
+        if (mRefreshType==REFRESH_ON_PULL || starredFiles == null || refreshTimeout)  {
```
Comment (Logan676) [inline src/com/seafile/seadroid2/ui/fragment/StarredFragment.java]: the refresh button on action bar does\`t work any more, change if condition like this
```
if (mRefreshType == REFRESH_ON_PULL
                || mRefreshType == REFRESH_ON_OVERFLOW_MENU
                || starredFiles == null
                || refreshTimeout)  {
```

**Comment by Logan676** [comment]
database_screenshot
@lins05 the starred files were cached in database correctly, so we can merge this after the review ticket was resolved.

**Comment by Logan676** [comment]
@forouher please resolve the review comment in order to merge this patch. Thanks.

**Pull request merged**